### ISSUE TITLE
Remote-WSL incorrectly considers all paths under /mnt/ to be part of the windows filesystem

### ISSUE THREAD

**Issue by tpyle** [microsoft/vscode-remote-release#5913]
- VSCode Version: 1.62.2
- Local OS Version: Windows 10 Pro 19043.1237
- Remote OS Version: Ubuntu 20.04.3 LTS
- Remote Extension/Connection Type: WSL
- Logs: N/A

Steps to Reproduce:

1. From WSL, create a path under the `/mnt/` directory (e.g. `/mnt/example`
2. Open that path in VS Code
3. Remote - WSL will produce a warning saying "This workspace is on the Windows file system (/mnt/). For best performance, we recommend moving the workspace to the Linux file system (~/home)"

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

The `/mnt/` directory is part of the linux filesystem:
```
$ df /mnt
Filesystem     1K-blocks    Used Available Use% Mounted on
/dev/sdb       263174212 7367944 242368112   3% /
```

The error message should be based on the device of the opened workspace, not the path.

**Comment by bamurtaugh** [comment]
Thanks for opening @typle! 

@aeschli If my understanding is correct (i.e. based on explanation [here](https://docs.microsoft.com/en-us/windows/wsl/filesystems)), `/mnt` is accessible from WSL, but it's a mounted drive from the Windows file system, and this message is correct, regardless of how the user accesses it. Or am I misunderstanding?

**Comment by aeschli** [comment]
You are right, not all mounts under `/mnt/` have to be Windows mounts.

I tuned the regex we used to also check for a drive letter. The fix will be in the next release of Remote-WSL (0.63.4)
Admittedly, that's still not 100% correct as one could use the same naming pattern for a manual mount. Also users can customize the name of the windows mount root in `/etc/wsl.conf`
But I think that's good enough for the purpose and I'd like to keep this check fast and simple for now.
